app/src/androidTest/java/org/piwigo/ui/launcher/LoginTest.java

```diff
@@ -0,0 +1,126 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.piwigo.R;
+import org.piwigo.ui.main.MainActivity;
+
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.pressImeActionButton;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.contrib.DrawerActions.open;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.CoreMatchers.allOf;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class LoginTest {
+
+    @Rule
+    public ActivityScenarioRule<LauncherActivity> activityScenarioRule =
+        new ActivityScenarioRule<LauncherActivity>(LauncherActivity.class);
+
+    @Test
+    public void loginTest() {
+
+        try {
+            Thread.sleep(1700);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ViewInteraction editUser = onView(withId(R.id.username));
+        editUser.check(matches(withText("")));
+
+        ViewInteraction editURL = onView(withId(R.id.url));
+        editURL.perform(replaceText("https://tg1.kulow.org"), closeSoftKeyboard());
+
+        editUser.perform(replaceText("seessmall"), closeSoftKeyboard());
+        editUser.perform(pressImeActionButton());
+
+        ViewInteraction editPassword = onView(withId(R.id.password));
+        editPassword.perform(replaceText("seessmall"), closeSoftKeyboard());
+        editPassword.perform(pressImeActionButton());
+
+        ViewInteraction loginButton = onView(allOf(withId(R.id.login_button), withContentDescription("Login")));
+        loginButton.perform(scrollTo(), click());
+
+        // Added a sleep statement to match the app's execution delay.
+        // The recommended way to handle such scenarios is to use Espresso idling resources:
+        // https://google.github.io/android-testing-support-library/docs/espresso/idling-resource/index.html
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ViewInteraction toolbar = onView(allOf(withText("Albums"), isDisplayed()));
```

Comment:
"Albums" is a localized text, isn't it? How do we want/need/plan to handle localization in the tests? Are test simply assumed to be executed in English environment?

Reply:
I had to switch my phone to english, yes.

We can also plainly skip this and only evaluate the presence of an id.

Reply:
And there are a lot of expectations on the environment - e.g. the app has to be freshly installed. I'm actually quite suprised on how raw it feels to do android testing.

Reply:
I guess it is fine to have that assumption because we don't need to test the localization code.
On the other side we'd like to keep testing as simple as possible to actually run them often, so checking for the id could be sufficient and more generally working.